src/core/query/ask.pl

```diff
@@ -173,15 +173,28 @@
 askable_prefixes(Descriptor,Prefixes) :-
     collection_descriptor_prefixes(Descriptor, Prefixes).
 
-create_context(Layer, Context) :-
+/**
+ * create_context(+Askable, -Context) is semidet.
+ *
+ * Create a context. Fail if Askable does not exist.
+ */
+create_context(Askable, Context) :-
+    ground(Askable),
+    var(Context),
+    !,
+    create_context_(Askable, Context).
+create_context(Askable, Context) :-
+    throw(error(unexpected_argument_instantiation(create_context, Askable, Context), _)).
```

Comment:
create_context_ may fail for various reasons, many of which have nothing to do with the groundless or varness of the input/output. Therefore fallthrough case catches too much.
I think it'd be better to use a `die_if` with the ground/var checks.

Reply:
You're right. I had that thought before I started, but I lost it somewhere in the middle.